### ISSUE TITLE
Propagate transaction pool error to `submit_at` in `OffchainSubmitTransaction`

### ISSUE THREAD

**Issue by armyhaylenko** [paritytech/substrate#11953]
Hey, I've been using the `SubmitTransaction::submit_unsigned_transaction` API, and whenever it fails, I just get a log like that:
`(offchain call) Error submitting a transaction to the pool: Transaction pool error`.
I feel like it would be useful to propagate this error to `submit_at` in `OffchainSubmitTransaction`, and from there - to `frame_system::offchain::SubmitTransaction::submit_transaction`, since we basically get no info about the error from the logs.

**Comment by bkchr** [comment]
Is this really the full output?
```
(offchain call) Error submitting a transaction to the pool: Transaction pool error
```

There should a reason following the `Transaction pool error`. 

Forwarding the entire error to the runtime would be a little bit complicated from a perspective of staying compatible over time, aka the error type could not change that easily. There would also be the question on what you would do from the runtime? You can not really handle the error and it just shows that you have some general bug with your implementation.

**Comment by armyhaylenko** [comment]
Yes, it's all. Everything would be fine if we could see the actual reason - but both in Native and in Wasm I can't see the error:

`2022-08-01 12:53:44.986  WARN tokio-runtime-worker txpool: [Parachain] (offchain call) Error submitting a transaction to the pool: Transaction pool error    `

**Comment by nazar-pc** [comment]
I believe I was fixing a similar problem in https://github.com/paritytech/substrate/pull/11629, are you on recent enough version of Substrate?

**Comment by armyhaylenko** [comment]
Hey, yeah, I'm on `polkadot-v0.9.24`. I think these changes are included in that branch. Let me clone the latest node template & see if that works.

**Comment by nazar-pc** [comment]
0.9.24 was out 5 days after that PR was merged, maybe 0.9.24 branched out without that fix. Try upgrading.

**Comment by armyhaylenko** [comment]
No, turns out this is available only from `polkadot-v0.9.25`... I think that closes the issue. Thanks guys.
